### PR TITLE
[TR][PAM] PIM-4729: Add ChoicesBuilder / UserContext cleaning

### PR DESCRIPTION
**Pull request by phaseinducer**
| Q                  | A
| ------------------ | ---
| Migration script?  | N
| Behats             | running
| Specs & Static     | Y
| Changelog updated  | Y

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Model/ChosableInterface.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ */
+interface ChosableInterface
+{
+    /**
+     * @return string
+     */
+    public function getChoiceValue();
+
+    /**
+     * @return string
```
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/Model/ChosableInterface.php]: missing phpdoc

File: src/Pim/Bundle/CatalogBundle/Model/ChosableInterface.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * Interface defining methods used in ChoicesBuilder to build choices lists
+ *
+ * @author    Yohan Blain <yohan.blain@akeneo.com>
+ * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+interface ChosableInterface
+{
+    /**
+     * @return string
```
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/Model/ChosableInterface.php]: missing phpdoc

File: src/Pim/Bundle/UserBundle/Context/UserContext.php [hunk 3]
```diff
@@ -31,46 +33,51 @@ class UserContext
...
-     * @param ChannelManager              $channelManager
+     * @param LocaleRepositoryInterface   $localeRepository
+     * @param ChannelRepositoryInterface  $channelRepository
      * @param CategoryRepositoryInterface $productCategoryRepo
      * @param RequestStack                $requestStack
      * @param string                      $defaultLocale
      */
     public function __construct(
         TokenStorageInterface $tokenStorage,
-        LocaleManager $localeManager,
```
Comment (skeleton) [inline src/Pim/Bundle/UserBundle/Context/UserContext.php]: BC break

**Comment by jjanvier** [comment]
GTM

File: src/Pim/Bundle/CatalogBundle/Resources/config/builders.yml [hunk 4]
```diff
@@ -1,6 +1,8 @@
 parameters:
     pim_catalog.builder.product.class:          Pim\Bundle\CatalogBundle\Builder\ProductBuilder
     pim_catalog.builder.product_template.class: Pim\Bundle\CatalogBundle\Builder\ProductTemplateBuilder
+    pim_catalog.builder.choices.class:          Pim\Bundle\CatalogBundle\Builder\ChoicesBuilder
+
```
Comment (nidup) [inline src/Pim/Bundle/CatalogBundle/Resources/config/builders.yml]: extra line

**Pull request merged**